src/group_theory/perm/cycles.lean

```diff
@@ -286,6 +303,37 @@ begin
   { exact (hb (extend_domain_apply_not_subtype _ _ pb)).elim }
 end
 
+lemma nodup_of_pairwise_disjoint_cycles {l : list (perm β)} (h1 : ∀ (f ∈ l), is_cycle f)
+  (h2 : l.pairwise disjoint) : l.nodup :=
+nodup_of_pairwise_disjoint (λ h, (h1 1 h).ne_one rfl) h2
+
+lemma pow_inj_on_lt_order_of (f : perm β) (n m : ℕ) (hn : n < order_of f) (hm : m < order_of f) :
+  f ^ n = f ^ m ↔ n = m :=
+begin
```

Comment:
Is it worth adding this statement, if we don't already have it, and using it to prove your statement? (or a similar one with `%`, I don't know which spelling is prefered)
```lean
lemma pow_eq_if_mod_eq (f : perm β) (n m : ℕ) (hn : n < order_of f) (hm : m < order_of f) :
  f ^ n = f ^ m ↔ n ≡ m [MOD (order_of f)] :=
```